src/lib.rs

```diff
@@ -1,21 +1,96 @@
 //! The project that will henceforth be known as `starling`.
-
+//!
+// `error_chain!` can recurse deeply
+#![recursion_limit = "1024"]
 #![deny(missing_docs)]
 #![deny(missing_debug_implementations)]
+#![deny(unsafe_code)]
+// Annoying warning emitted from the `error_chain!` macro.
+#![allow(unused_doc_comment)]
 
+extern crate clap;
+#[macro_use]
+extern crate error_chain;
 extern crate futures;
 extern crate tokio_core;
 
+pub mod cli;
```

Comment:
You want to move the `cli` mod into `src/bin/starling.rs`, or you don't want the module to be `pub`, or you want it to be in a whole new create? It's not clear to me what you're requesting here.